src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -374,14 +379,15 @@ public void fireFileStarted(String fileName) {
      * Notify all listeners about the errors in a file.
      *
      * @param fileName the audited file
-     * @param errors the audit errors from the file
+     * @param fileProcessingResult file processing results object
      */
     @Override
-    public void fireErrors(String fileName, SortedSet<LocalizedMessage> errors) {
+    public void fireErrors(String fileName, FileProcessingResult fileProcessingResult) {
```

Comment:
@MEZk @rnveach 
I'm stuck with one thing. Inside [Checker#processFile,](https://github.com/timurt/checkstyle/commit/8da878eaf14710f1d578ecadfa1caafe94512dc3#diff-503cc093d3169abd5697cc8189f27f32R348) there is try-catch for `IOException` in a case specified does not exist, it means message will be logged, but `FileText` will be null. If FileText object is null then we will not be able to extract `fileName`, `absolutePath` and etc. from it, for example inside `Checker#fireErrors` it will throw `NPE`.
Problem is `fileName` is used in various places even if the file does not exist
I can suggest to add additional `filePath` field to `FileProcessingResult` or left PRed implementation?